src/vsr/replica.zig

```diff
@@ -5278,6 +5280,14 @@ pub fn ReplicaType(
 
             if (self.status == .view_change and self.primary_index(self.view) == self.replica) {
                 if (self.journal.dirty.count == 0 and self.commit_min == self.commit_max) {
+                    if (self.commit_stage != .idle) {
+                        // If we still have a commit running, we started it the last time we were
+                        // primary, and its still running. Wait for it to finish before repairing
+                        // the pipeline so that it doesn't wind up in the new pipeline.
+                        assert(self.commit_prepare.?.header.op == self.commit_min + 1);
```

Comment:
Assert that prepare's view is strictly less than our current view.

It would be cool to also assert that we were indeed a primary when we started commiting this, but I think we can'd just check that primary_index(prepare.view) == self.replica, because we might be committing someone else's prepare from a view before.